### DISCUSSION TITLE
Плагины в виде .so/.dylib

### DISCUSSION THREAD

**Discussion by pg83** [elfmz/far2l#1093] [General]
Привет.

А зачем встроенные плагины сделаны в виде загружаемых модулей? Это мешает собрать far для статически линкуемых дистрибутивов. Я мог бы сделать так, чтобы все плагины из репозитория far2l линковались в бинарник far статически. Я посмотрел, это должно быть не очень сложно.

Тут возникают следующие вопросы:

1) Предполагается ли, что far2l будет работать под винду?
2) Нужно ли поддерживать возможность собирать встроенные плагины в виде .so, в дополнение к статической сборке? Если нужно, то я не настолько хорошо знаю CMake.

PS: я бы очень вам советовал прогнать clang-format на исходники, если не предполагается это все мержить с upstream. Читать такой разносортный и разностильный код очень тяжело.

**Comment by pg83** [comment]
@elfmz

**Comment by elfmz** [comment]
Плагины - ну так ведь повелосьс виндового фара, по поводу статически собрать - так просто это сделать не получится - ведь каждый плагин имеет набор экспортируемых функций и они у разных плагинов одинаковые имена имеют. Плюс логика со стороны фара тоже подразумевает некий набор объектов плагинов. Хотя в принципе все это решаемо если прям так уж надо - можно для статической линковки префиксить экспорты каждого из плагинов его уникальным именем, потом собрать их в структурки и сами структуры - в массивчик. Нечто такое я делал в мультиарке - ведь оригинальный мультиарк имел для обработчков разных форматов так сказать 'субплагины' - отдельная длл для zip, отдельная для rar и т.п.

1 Был некогда эксперимент со сборкой под Cygwin, работало, но серъезно это никогда не поддерживалось, но и не ломалось осознанно. А почему возник такой вопрос?
2 Врядли. Динамические плагины как таковые - виндовое наследие малоосмысленное в мире опенсорса, хотя в линуксе от них таки тоже может быть польза - через apt ставить по-отдельности то-се. Плюс плагиновая архитектура дисциплинирует разработку. А так..., для far2l насколько я знаю есть только один отдельно написанный 3rd-party плагин - https://github.com/cycleg/far-gvfs/ решайте сами насколько он вам важен.

Да переформатить тоже всегда хотелось, но со временем свыкся. В этом есть свои плюсы - сразу видно где какой код, и что от него можно ожидать :) Но может какнить попробую все под одну гребенку и если понравится - вкомичу.
